### PR TITLE
Cookies 2024 chapter

### PR DESCRIPTION
**Pull request by ydimova**
Fixes: #3617

[Staged](https://cookies-2024-dot-webalmanac.uk.r.appspot.com/en/2024/cookies)
Featured chapter home page link: https://cookies-2024-dot-webalmanac.uk.r.appspot.com/en/2024/?feat=cookies

### CODE REVIEW

**Comment by tunetheweb** [comment]
I staged the chapter here: https://cookies-2024-dot-webalmanac.uk.r.appspot.com/en/2024/cookies

I had to enable the chapter in `src/config/2024.json` (could you add that to this PR please?) and also fix some markup:

```diff
diff --git a/src/config/2024.json b/src/config/2024.json
index a1abf227..3bf89f77 100644
--- a/src/config/2024.json
+++ b/src/config/2024.json
@@ -199,8 +199,7 @@
           "part": "IV",
           "chapter_number": "24",
           "title": "Cookies",
-          "slug": "cookies",
-          "todo": true
+          "slug": "cookies"
         }
       ]
     }
diff --git a/src/content/en/2024/cookies.md b/src/content/en/2024/cookies.md
index f9f6fe5b..75c4cb66 100644
--- a/src/content/en/2024/cookies.md
+++ b/src/content/en/2024/cookies.md
@@ -160,7 +160,8 @@ Additionally,  the prevalence of each cookie type across the ranks is quite simi

 Next, we discuss the distribution of different cookie [attributes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie). Furthermore, we zoom into the use of the "SameSite" cookie attribute. The following two figures show the proportion of first and third-party cookies set on the top 1M websites for each client that have one of the following attributes set: "Partitioned", "Session", "HttpOnly", "Secure", "SameSite". Before diving into more details for each attribute, let's observe here again the similarity of the distribution of the different attributes between  desktop or mobile clients.

-{{ figure_markup(https://docs.google.com/spreadsheets/d/e/2PACX-1vTLO9Te80QewkuPKnz6eJ7OFcU5q3fZMsdqv7cEncBKrL5zcsIN9sMMg5HQT7ndKze8JJNe-V1IkB-9/pubchart?oid=2023103422&format=interactive
+{{ figure_markup(
+  image="cookie-attributes.png",
   description="This figures gives an overview of how cookie attributes are used for desktop clients for both first- and third-party cookies. 100% of third-party cookies include the SameSite and Secure attributes. Only 1% of first-party cookies and 6% of third-party cookies use partioned. 16% of first-party cookies set their Session attribute, while this is the case for only 4% of third-party cookies. Finally, 12% of first-party cookies and 19% of third-party cookies use the HttpOnly attribute.",
   chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTLO9Te80QewkuPKnz6eJ7OFcU5q3fZMsdqv7cEncBKrL5zcsIN9sMMg5HQT7ndKze8JJNe-V1IkB-9/pubchart?oid=2023103422&format=interactive",
   sheets_gid="1700493344"
@@ -300,7 +301,7 @@ In the following section, we report for each client (desktop and mobile) the top
 {{ figure_markup(
   image="top-first-party-cookies-set.png",
   caption="Top first-party cookies set",
-  description="The chart shows the most widely-set first-party cookies. Google analytics sets the "_ga" and "_gid" cookies, which are used for website statistics and analytics reports, on more than 61% of websites fot both mobile and desktop clients." ,
+  description="The chart shows the most widely-set first-party cookies. Google analytics sets the `_ga` and `_gid` cookies, which are used for website statistics and analytics reports, on more than 61% of websites fot both mobile and desktop clients." ,
   chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTLO9Te80QewkuPKnz6eJ7OFcU5q3fZMsdqv7cEncBKrL5zcsIN9sMMg5HQT7ndKze8JJNe-V1IkB-9/pubchart?oid=380182537&format=intera
ctive",
   sheets_gid="1236728722",
   sql_file = 'top_20_first_party_cookies.sql'
@@ -810,9 +811,9 @@ See the [Privacy chapter](https://almanac.httparchive.org/en/2024/privacy) for s

 {{ figure_markup(
   caption="Number of related primary website sets.",
-  content= 64 related primary website sets validated by Google at the moment",
+  content="64 related primary website sets validated by Google at the moment",
   classes="really-big-number",
-  sheets_gid="199073475""
+  sheets_gid="199073475"
 )
 }}

@@ -823,9 +824,10 @@ Chrome ships with a pre-loaded file containing related website sets validated by
 {{ figure_markup(
   image="secondary-domains.png",
   caption="Secondary domains per primary domain.",
-  description="The graph shows secondary domains associated to primary domains for Related Website Sets, which is part of Google's Privacy Sandbox. We observe that if a majority of the primary domains are associated with 5 or less secondary domains, "https://journaldesfemmes.com", "https://ya.ru", and "https://mercadolibre.com" are linked to 8, 17, and 39 secondary domains among which third party requests are handled as if they were all from the first party, respectively." ,
+  description="The graph shows secondary domains associated to primary domains for Related Website Sets, which is part of Google's Privacy Sandbox. We observe that if a majority of the primary domains are associated with 5 or less secondary domains, `https://journaldesfemmes.com`, `https://ya.ru`, and `https://mercadolibre.com` are linked to 8, 17, and 39 secondary domains among which third party requests are handled as if they were all from the first party, respectively.",
   chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTLO9Te80QewkuPKnz6eJ7OFcU5q3fZMsdqv7cEncBKrL5zcsIN9sMMg5HQT7ndKze8JJNe-V1IkB-9/pubchart?oid=627102878&format=interactive",
-  sheets_gid="199073475"  )
+  sheets_gid="199073475"
+)
 }}
```

**Comment by tunetheweb** [comment]
Next set of changes would be to add the contributors to the `src/config/contributors.json` file (alphabetical by first name if you can please!)

Here's a set of diffs for you if that's helpful (though I don't have @turban1988's real name if they want that included):

```diff
--- a/src/config/contributors.json
+++ b/src/config/contributors.json
@@ -917,6 +917,17 @@
     },
     "website": "https://chrisadams.me.uk"
   },
+  "chrisbeety": {
+    "avatar_url": "32492572",
+    "github": "chrisbeety",
+    "name": "Chris Beety",
+    "teams": {
+      "2024": [
+        "analysts"
+      ]
+    },
+    "website": "https://chrisadams.me.uk"
+  },
   "svgeesus": {
     "avatar_url": "2506926",
     "github": "svgeesus",
@@ -3557,6 +3568,9 @@
       "2019": [
         "brainstormers",
         "authors"
+      ],
+      "2024": [
+        "authors"
       ]
     },
     "twitter": "sw12",
@@ -3705,6 +3719,16 @@
       ]
     }
   },
+  "shaoormunir": {
+    "avatar_url": "22539212",
+    "github": "shaoormunir",
+    "name": "Shaoor Munir",
+    "teams": {
+      "2024": [
+        "authors"
+      ]
+    }
+  },
   "spanicker": {
     "avatar_url": "4945616",
     "github": "spanicker",
@@ -4056,6 +4080,16 @@
       ]
     }
   },
+  "turban1988": {
+    "avatar_url": "29097632",
+    "github": "turban1988",
+    "name": "Turban",
+    "teams": {
+      "2024": [
+        "reviewers"
+      ]
+    }
+  },
   "TusharPol": {
     "avatar_url": "104352788",
     "github": "TusharPol",
@@ -4227,6 +4261,9 @@
       ],
       "2022": [
         "analysts"
+      ],
+      "2024": [
+        "authors"
       ]
     }
   },
@@ -4246,6 +4283,17 @@
     "twitter": "yoavweiss",
     "website": "https://blog.yoav.ws"
   },
+  "yohanbeugin": {
+    "avatar_url": "32905060",
+    "github": "yohanbeugin",
+    "name": "Yohan Beugin",
+    "teams": {
+      "2024": [
+        "analysts"
+      ]
+    },
+    "website": "https://yohan.beugin.org"
+  },
   "tyohan": {
     "avatar_url": "106703",
     "github": "tyohan",
diff --git a/src/content/en/2024/cookies.md b/src/content/en/2024/cookies.md
index a91f706d..d28acbe4 100644
--- a/src/content/en/2024/cookies.md
+++ b/src/content/en/2024/cookies.md
@@ -3,7 +3,7 @@
 title: Cookies
 description: Cookies chapter of the 2024 Web Almanac covering the prevalence and structure of cookies on the web.
 authors: [shaoormunir,samdutton,ydimova]
-reviewers: [turban]
+reviewers: [turban1988]
 analysts: [chrisbeety, yohanbeugin]
 editors: []
 translators: []
```

**Comment by yohhaan** [comment]
Hello everyone,

I finished a formatting pass on the markdown conversion of the chapter that @ydimova had started to work on. Summary of changes:
- [x] Correct list of contributors to chapter updated + corresponding modifications to contributors file
- [x] Typo & Formatting pass
- [x] All figures were re-created and re-published to be of the default size (I was not aware of that requirement when I initially generated them over the summer  :cry: )
- [x] Correct SQL queries, sheet gid, etc,. listed for figures

Thanks @tunetheweb for helping with the linting issues: 
- only one check not passing, but not sure what it is about, so I may defer to you? 
- also, we refer in our chapter to other chapters (like privacy and security 2024), what is the best way for the hyperlink here? (I put the complete url where I assume it is going to be published, but should it be a relative one maybe?)
- is there a methodology chapter for 2024 explaining how the crawl are ran? (we refer to one based on the fact that there was one in the past, but not sure for 2024)

Thanks! I will take some more time tomorrow to read over the cookies 2024 chapter again tomorrow and fix issues (if any)
Good night!

**Comment by tunetheweb** [comment]
> * only one check not passing, but not sure what it is about, so I may defer to you?

It’s complaining about the chapter hero image being missing. I contacted a designer friend last week and hoping to have that for launch. If not we’ll just have to use the Privacy one for a bit.

> * also, we refer in our chapter to other chapters (like privacy and security 2024), what is the best way for the hyperlink here? (I put the complete url where I assume it is going to be published, but should it be a relative one maybe?)

Yes relative links. That way they’ll be relative for any translations too. For example; (`the [Security](./security) chapter` or `the [2022 Security](../2022/security) chapter`.

> * is there a methodology chapter for 2024 explaining how the crawl are ran? (we refer to one as there was one in past editions, just not sure for 2024)

Yup. Already merged a few weeks back. Update any references to that.

**Comment by yohhaan** [comment]
Thanks for the answer!

- I have added the compressed static figures.
- I have also  completed my final pass on the converted markdown, everything should be fine: I can also check again if the staged version of the chapter gets updated. 

Ping me if there is something else to be done for getting the chapter ready to be published.
Thanks!
Best,

**Comment by tunetheweb** [comment]
Restaged it. Will take a proper pass through tomorrow before we release.

File: src/content/en/2024/cookies.md [hunk 1]
```diff
@@ -1,16 +1,820 @@
...
+- Data is collected by automatically visiting websites in a non-interactive way; user interaction could modify the way websites set and use cookies in practice. For example, HTTP Archive's tools do not interact with cookie banners (if any) and so cookies that would be set after interaction with these banners are not observed by our study.
+- Websites are visited from servers located in the US that have no cookie set when each independent website visit starts; this is quite different from a user accumulating and saving web cookies while browsing the web. The location from which visits are performed can impact cookie behavior due to regulation and legislation such as [GDPR](https://gdpr-info.eu/).
+- For each website, the home page is visited as well as one other page from the same website.
+- Most of the results presented in this chapter are based on the top one million (top 1M by CrUX rank) most visited websites that were successfully reached during the HTTP Archive crawl of June 2024.
```
Comment (tunetheweb) [inline src/content/en/2024/cookies.md:107]: This is no longer true is it? Think we reran for ALL the sites. So I'll update this in the final edit I'm doing now.

**Review by tunetheweb — APPROVED** [review]

File: src/content/en/2024/cookies.md (hunk 1 above)
Comment (yohhaan) [reply src/content/en/2024/cookies.md]: Yes, leave as is: 
- I only regenerated the first figures 1st/3rd party prevalence across all ranks
- I would not have had enough time to re-extract all stats, re-plot everything, and update the text content while also formatting and reviewing the chapter for today. 
- Given that the top 1M websites capture about 95% of all page loads of web users according to [this measurement paper](https://dl.acm.org/doi/pdf/10.1145/3517745.3561418) on Google's private telemetry dataset, I focused on finalizing what we had and left a comment for future editions to maybe run the other queries not only on top 1M websites. 

Thanks!

**Pull request merged**